File: Python/14.longest-common-prefix.py
#
# @lc app=leetcode id=14 lang=python3
#
# [14] Longest Common Prefix
#
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        StrsLen = len(strs)
        if StrsLen == 0:
            return ''
        
        SortedStrs = sorted(strs, key=lambda x:len(x))
        prefix = SortedStrs[0]
        for i in range(1, StrsLen):
            while SortedStrs[i].find(prefix) != 0:
                prefix = prefix[:len(prefix) - 1]
                if prefix == '':
                    return prefix
        return prefix

File: Java/14.longest-common-prefix.java
/*
 * @lc app=leetcode id=14 lang=java
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }

        String result = strs[0];
        for (int i = 1; i < strs.length; i++) {
            // find the shortest str
            if (strs[i].length() < result.length()) {
                result = strs[i];
            }
        }
        for (int i = 0; i < strs.length; i++) {
            while (strs[i].indexOf(result) != 0) {
                result = result.substring(0, result.length() - 1);
            }
        }
        
        return result;
    }
}
// @lc code=end

